ExampleMod/Items/ExampleDashAccessory.cs

```diff
@@ -0,0 +1,208 @@
+			// If the Shield of Cthulhu, Master Ninja Gear, Tabi and/or Solar Armour set is equipped, prevent this accessory from doing its dash effect
+
+			//The priority is used to prevent undesirable effects.
+			//Without it, the player is able to use the ExampleDashAccessory's dash as well as the vanilla ones
+			bool dashAccessoryEquipped = false;
+			bool shieldOfCthulhuEquipped = false;
+			bool masterNinjaGearEquipped = false;
+			bool tabiEquipped = false;
+
+			//This is the loop used in vanilla to update/check the not-vanity accessories
+			for(int i = 3; i < 8 + player.extraAccessorySlots; i++)
+			{
+				Item item = player.armor[i];
+
+				//Set the corresponding flag for each item if that item is equipped
+				if(item.type == ItemType<ExampleDashAccessory>())
+				{
+					dashAccessoryEquipped = true;
+				}
+				else if(item.type == ItemID.EoCShield) {
+					shieldOfCthulhuEquipped = true;
+				}
+				else if(item.type == ItemID.MasterNinjaGear) {
+					masterNinjaGearEquipped = true;
+				}
+				else if(item.type == ItemID.Tabi) {
+					tabiEquipped = true;
+				}
+			}
+
+			//If we don't have the ExampleDashAccessory equipped or if any of the higher-priority accessories/armour mentioned above are equipped, return immediately
+			//Also return if the player is currently on a mount, since dashes on a mount look weird
+			if(!dashAccessoryEquipped || shieldOfCthulhuEquipped || masterNinjaGearEquipped || tabiEquipped || player.setSolar || player.mount.Active)
+			{
+				return;
+			}
+
+			//Set the "pressed" bools
+			PressedDown = player.controlDown && player.releaseDown;
+			PressedUp = player.controlUp && player.releaseUp;
+			PressedRight = player.controlRight && player.releaseRight;
+			PressedLeft = player.controlLeft && player.releaseLeft;
+
+			//Finally, if any of the "DoubleTap" properties are true and the dash hasn't started, set the dash to being active
+			//Examples for alternate checks are the following:
+			// Right/Left only:  if(!DashActive && (DoubleTapRight || DoubleTapLeft))
+			// Up only:          if(!DashActive && DoubleTapUp)
+			if(!DashActive && (DoubleTapDown || DoubleTapUp || DoubleTapRight || DoubleTapLeft))
+			{
+				DashActive = true;
+
+				//Here you'd be able to set an effect that happens when the dash first activates
+				//Some examples include:  the larger smoke effect from the Master Ninja Gear and Tabi
+			}
+		}
+
+		public override void PostUpdate()
+		{
+			//This hook will be used to assign the values in our oldCardinalTimers array to the player.doubleTapCardinalTimer array
+
+			//Check if each direction was pressed.  If it was, store the old value
+			//Using a helper method to shorten repetition
+			UpdateTimer(Down,  PressedDown);
+			UpdateTimer(Up,    PressedUp);
+			UpdateTimer(Right, PressedRight);
+			UpdateTimer(Left,  PressedLeft);
+		}
+
+		private void UpdateTimer(int timerIndex, bool condition)
+		{
+			if(condition)
```

Comment:
The way you detect double taps isn't very well explained, I had to consult the terraria source to understand how it's supposed to work. By my read:

`ResetEffects` is called after the vanilla `doubleTapCardinalTimer` logic for a tick. On the first press-release, the timer is set to 15. At this point your `condition` will be true here, and you'll copy the value `15` into `oldCardinalTimers`.

Vanilla gives 15 ticks for the second tap (1/4 of a second) by counting down this variable. It doesn't reset either, so it looks to me like you could just use the same `doubleTapCardinalTimer[l] > 0` check directly.

Your approach of checking if the current doubletap timer is > then the one when the key was pressed the first time (`oldCardinalTimers`), and then clearing it when it reaches zero works, but you're effectively just doing `player.doubleTapCardinalTimer < 15 && player.doubleTapCardinalTimer > 0`
the first of which will always be true